packages/components/psammead-content-shift-blocker/src/index.stories.jsx

```diff
@@ -0,0 +1,173 @@
+];
+const renderParagraphs = ({ script, service }) =>
+  paragraphs.map(paragraph => (
+    <Paragraph script={script} service={service}>
+      {paragraph}
+    </Paragraph>
+  ));
+
+storiesOf('Components|ContentShiftBlocker', module)
+  .addDecorator(withKnobs)
+  .addDecorator(withServicesKnob())
+  .add(
+    'vertical content shift',
+    ({ script, service }) => {
+      return (
+        <Wrapper>
+          <Headline>The Daily Doggo</Headline>
+          {renderParagraphs({ script, service })}
+          <Gallery script={script} service={service} />
+          {renderParagraphs({ script, service })}
+          {renderParagraphs({ script, service })}
+        </Wrapper>
+      );
+    },
+    { notes },
+  )
+  .add(
+    'multiple',
+    ({ script, service }) => {
+      return (
+        <Wrapper>
+          <Headline>The Daily Doggo</Headline>
+          {renderParagraphs({ script, service })}
+          <Gallery script={script} service={service} />
+          {renderParagraphs({ script, service })}
+          <Gallery script={script} service={service} />
+          {renderParagraphs({ script, service })}
+          {renderParagraphs({ script, service })}
+          {renderParagraphs({ script, service })}
+          <Gallery script={script} service={service} />
+          <Gallery script={script} service={service} />
+          {renderParagraphs({ script, service })}
+          {renderParagraphs({ script, service })}
+          {renderParagraphs({ script, service })}
+          {renderParagraphs({ script, service })}
+        </Wrapper>
+      );
+    },
+    { notes },
+  )
+  .add(
+    'multiple staggered',
+    ({ script, service }) => {
+      return (
+        <Wrapper>
+          <Headline>The Daily Doggo</Headline>
+          {renderParagraphs({ script, service })}
+          <Gallery delay={4000} script={script} service={service} />
+          {renderParagraphs({ script, service })}
+          <Gallery delay={3000} script={script} service={service} />
+          {renderParagraphs({ script, service })}
+          {renderParagraphs({ script, service })}
+          {renderParagraphs({ script, service })}
+          <Gallery delay={2000} script={script} service={service} />
+          <Gallery delay={1000} script={script} service={service} />
+          {renderParagraphs({ script, service })}
+          {renderParagraphs({ script, service })}
+          {renderParagraphs({ script, service })}
+          {renderParagraphs({ script, service })}
+        </Wrapper>
+      );
+    },
+    { notes },
+  )
+  .add(
+    'nothing but cats',
```

Comment:
I've only been able to replicate this bug in Firefox at the moment - when viewing this story, scrolling down half-way and waiting for about 30 seconds, something seems to be going out of sync, because the boxes jump around the page.

<img src="" alt="Gif of 'nothing but cats' story with large jumps in the page" />

Reply:
@sareh I changed the scroll position adjustment to happen in `useLayoutEffect` instead of `useEffect` and that has fixed this issue.